flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/library/LabelPropagation.java

```diff
@@ -68,12 +70,12 @@ public LabelPropagation(int maxIterations) {
 	public static final class UpdateVertexLabel<K extends Comparable<K> & Serializable>
 			extends VertexUpdateFunction<K, Long, Long> {
 
-		public void updateVertex(K vertexKey, Long vertexValue,
+		public void updateVertex(Vertex<K, Tuple3<Long,Long,Long>> vertex,
 				MessageIterator<Long> inMessages) {
 			Map<Long, Long> labelsWithFrequencies = new HashMap<Long, Long>();
 
 			long maxFrequency = 1;
-			long mostFrequentLabel = vertexValue;
+			long mostFrequentLabel = vertex.getValue().f0;
```

Comment:
`vertex.getValue()` should return the vertex value, not a Tuple3 from the user has to extract the vertex value :S